rfcs/0120-artifact-service.md

```diff
@@ -0,0 +1,296 @@
+# RFC 120 object Service
+* Comments: [#116](https://api.github.com/repos/taskcluster/taskcluster-rfcs/issues/120)
+* Proposed by: @jhford
+
+# Summary
+
+Taskcluster tasks often produce artifacts during the course of being run.
+Examples include browser distribution archives, json metadata and log files.
+We currently store artifacts in a single region of S3 and split management of
+artifacts between the Queue and Cloud Mirror.  This project is to design an
+object storage api which drastically simplifies the management of objects from
+the Queue's perspective while at the same time handling the cross-region
+replication and cache management of Cloud Mirror as well as enabling
+cross-cloud support.
+
+The object service should not be directly exposed to end users as a Rest API.
+It will instead be a generalised API that the Queue uses to facilitate artifact
+creation from tasks.  In other words, the Queue gates operations to the object
```

Comment:
Am I correct in understanding that artifact persistence is the primary use case of the object service, but it is not limited to storing artifacts (so other taskcluster services might use it for persisting other types of data)? If this is the case, I think the name "Object Service" makes sense, and in the summary we should highlight that artifact persistence is the primary use case of this service, but that it is a general purpose binary storage interface, rather than being artifact-specific. If it is an artifact-specific service, I think we should rename the RFC to "Artifact Service".

Reply:
We discussed this at length in Berlin, and decided (or at least "[disagree and commit](http://www.amazonianblog.com/2016/11/what-it-means-to-disagree-and-commit-and-how-i-do-it.html)") that yes, it is not artifact-specific and could potentially host other objects.